### PR TITLE
add common implementation of geojson data source, with feature using …

### LINKED ISSUE #19: Implement Mapbox GeoJson data source for Android and JS
# Overview

Implement a GeoJson data source in the `mapbox` module's common code and then provide implementations for Android and JS. The implementations can be based off of the implementations in [mmapp](https://github.com/dellisd/mmapp/blob/master/shared/src/commonMain/kotlin/io/dellisd/github/mmapp/GeoJsonSource.kt).

# Technical Details

In common Kotlin code, we can declare a `GeoJsonSource` like so: `val source = GeoJsonSource(data)` .

When this common declaration is imported to platform-specific view (i.e. to use in the view to add to the Map), we would like to be able to treat the `source` object as though it were already a platform-specific instance of the data source. In other words, we would like to be able to say that our `source` **is a** `com.mapbox.mapboxsdk.style.sources.GeoJsonSource` when referenced in Android code, and that our `source` matches the contract for a data source for use in the mapbox-gl-js library. (i.e. has `type: "geojson"` and `data` properties). Examples of how this works can be found in the mmapp project.

# Task Instructions

1. Create the common definition of the `GeoJsonSource`.
2. Create `actual` implementations of this class for Android and JS.
3. Create an instance of a `GeoJsonSource` with some random feature and then add them to the map that was added in #15 and #17.

### PR DESCRIPTION
**Pull request by ghaskins99**
resolves #19.

### CODE REVIEW

**Review by WillMahler — COMMENTED** [review]
Looks good to me!

**Review by dellisd — APPROVED** [review]

File: common/build.gradle.kts [hunk 1]
```diff
@@ -89,6 +90,7 @@ kotlin {
     sourceSets["iosMain"].dependencies {
         implementation(deps.sqldelight.nativeDriver)
         implementation(deps.ktor.iosDriver)
+        implementation(deps.spatialk.geojson)
```
Comment (dellisd) [inline common/build.gradle.kts:93]:
```suggestion
```

File: common/build.gradle.kts [hunk 2]
```diff
@@ -79,6 +79,7 @@ kotlin {
         implementation(deps.sqldelight.androidDriver)
         api(deps.mapbox.androidSdk)
         implementation(deps.ktor.androidDriver)
+        api(deps.spatialk.geojson)
```
Comment (dellisd) [inline common/build.gradle.kts:82]:
```suggestion
```

**Review by dellisd — CHANGES_REQUESTED** [review]
These should actually be covered by the `api` call in the `commonMain` dependencies, no?

**Review by WillMahler — APPROVED** [review]

**Review by dellisd — APPROVED** [review]

**Review by dellisd — CHANGES_REQUESTED** [review]
Change the android implemnetation to use your new data source class

**Review by dellisd — CHANGES_REQUESTED** [review]
One last thing:
Can we move all mapbox-related code to the `mapbox` module (instead of `common`) and put it all in the `com.cuhacking.mapbox` package?

**Review by dellisd — APPROVED** [review]

**Pull request merged**